x-pack/plugins/security_solution/public/common/store/data_table/timeline.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get, omit } from 'lodash/fp';
+import type { Action } from 'redux';
+import type { Storage } from '@kbn/kibana-utils-plugin/public';
+import type { TimelineModel } from '../../../timelines/store/timeline/model';
+import type { TimelineById } from '../../../timelines/store/timeline/types';
+import {
+  applyKqlFilterQuery,
+  addProvider,
+  addTimeline,
+  dataProviderEdited,
+  removeProvider,
+  saveTimeline,
+  setExcludedRowRendererIds,
+  setFilters,
+  setSavedQueryId,
+  updateDataProviderEnabled,
+  updateDataProviderExcluded,
+  updateDataProviderType,
+  updateEqlOptions,
+  updateKqlMode,
+  updateProviders,
+  updateTitleAndDescription,
+  updateDataView,
+  removeColumn as timelineRemoveColumn,
+  updateColumns as timelineUpdateColumns,
+  updateSort as timelineUpdateSort,
+  updateRange as timelineUpdateRange,
+  upsertColumn as timelineUpsertColumn,
+  updateSavedSearchId,
+} from '../../../timelines/store/timeline/actions';
+
+export const timelineActionTypes = [
+  applyKqlFilterQuery.type,
+  addProvider.type,
+  addTimeline.type,
+  dataProviderEdited.type,
+  removeProvider.type,
+  saveTimeline.type,
+  setExcludedRowRendererIds.type,
+  setFilters.type,
+  setSavedQueryId.type,
+  updateDataProviderEnabled.type,
+  updateDataProviderExcluded.type,
+  updateDataProviderType.type,
+  updateEqlOptions.type,
+  updateKqlMode.type,
+  updateProviders.type,
+  updateTitleAndDescription.type,
+  updateDataView.type,
+  timelineRemoveColumn.type,
+  timelineUpdateColumns.type,
+  timelineUpdateSort.type,
+  timelineUpdateRange.type,
+  timelineUpsertColumn.type,
+
```

Comment:
Depending on which PR goes in first, @logeekal, can we add the `changed` action from @janmonschke's save PR?

Reply:
Since @janmonschke 's PR is almost in, I can wait for his to go in and then make changes.

Reply:
His PR merged, so if you can add those changes, and I'll add the approve. My other comments were primarily nits